processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingOptions.java

```diff
@@ -421,7 +421,7 @@ public FormattingParameters getFormattingParameters() {
     }
 
     public SelectionParameters getSelectionParameters() {
-        return selectionParameters;
+        return selectionParameters.isEmpty() ? null : selectionParameters;
```

Comment:
Is it "safe" to just return null here?

Reply:
This is no longer necessary, I will remove it with the next commit.